application/classes/Controller/Api/Posts/Lock.php

```diff
@@ -0,0 +1,61 @@
+<?php defined('SYSPATH') OR die('No direct access allowed.');
+
+/**
+ * Ushahidi API Forms Groups Controller
+ *
+ * @author     Ushahidi Team <team@ushahidi.com>
+ * @package    Ushahidi\Application\Controllers
+ * @copyright  2013 Ushahidi
+ * @license    https://www.gnu.org/licenses/agpl-3.0.html GNU Affero General Public License Version 3 (AGPL3)
+ */
+
+class Controller_API_Posts_Lock extends Ushahidi_Rest {
+
+	protected $_action_map = array
+	(
+		Http_Request::GET     => 'get',
+		Http_Request::POST    => 'post',
+		Http_Request::PUT     => 'put',
+		Http_Request::OPTIONS => 'options'
+	);
+
+	protected function _scope()
+	{
+		return 'posts';
+	}
+
+	protected function _resource()
+	{
+		return 'posts_lock';
+	}
+
+	// Check Lock
+	public function action_get_index_collection()
+	{
+		$this->_usecase = service('factory.usecase')
+			->get($this->_resource(), 'checkLock');
+		$this->_usecase			
+			->setIdentifiers($this->request->param())
+			->setFormatter(service("formatter.entity.post.check.lock"));
+	}
+
+	// Get Lock
+	public function action_post_index_collection()
```

Comment:
I think the verbs are wrong here. We could use PUT / DELETE to create / break locks, but POST / PUT.
Similarly PUT on an existing lock, for the same user should simply update the lock. That'd allow simple polling if the user has a post open.